mimic/model/lstm_model.py

```diff
@@ -0,0 +1,80 @@
+from mimic.model.model import Model
+
+from keras.preprocessing.sequence import pad_sequences
+from keras.layers import Embedding, LSTM, Dense, Dropout
+from keras.preprocessing.text import Tokenizer
+from keras.callbacks import EarlyStopping
+from keras.models import Sequential
+import keras.utils as ku 
+import tensorflow as tf
+
+import numpy as np
+import string, os 
+
+class LSTMModel(Model):
+
+    def __init__(self):
+        self.tokenizer = Tokenizer()
+
+    ''' Create sequences to train the model - each is a sequence of tokens that represent characters '''
+    def learn(self, text):
+
+        # TODO These are currently arbitrary - we can look at varying these depending on the size of the input text
+        SEQ_LEN = 100
+        BATCH_SIZE = 200
+
+        # Clean text of weird characters
+        text = "".join(v for v in text if v not in string.punctuation).lower()
+        clean_txt = text.encode("utf8").decode("ascii",'ignore')
```

Comment:
Do you think this would also be valuable for other models (e.g. Markov chain)? If so, we might want to refactor this into a cleaning/preprocessing module (that would presumably be called by the generator when loading texting).

Reply:
(That being said, I wouldn't bother changing it for this PR - we can make a new ticket to change it.)

Reply:
Yeah, most likely. I was thinking to do so but we haven't discussed how we were going to handle util methods yet